flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/connector/kafka/source/reader/KafkaPartitionSplitReaderTest.java

```diff
@@ -323,18 +323,19 @@ public void testUsingCommittedOffsetsWithEarliestOrLatestOffsetResetStrategy(
     @Test
     public void testConsumerClientRackSupplier() {
         AtomicReference<Boolean> supplierCalled = new AtomicReference<>(false);
+        String rackId = "use-az1";
         Supplier<String> rackIdSupplier =
                 () -> {
                     supplierCalled.set(true);
-                    return "foo";
+                    return rackId;
                 };
         Properties properties = new Properties();
         createReader(
                 properties,
                 UnregisteredMetricsGroup.createSourceReaderMetricGroup(),
                 rackIdSupplier);
         assertThat(supplierCalled.get()).isEqualTo(true);
-        assertThat("foo".equals(properties.getProperty(ConsumerConfig.CLIENT_RACK_CONFIG)));
+        assertThat(rackId.equals(properties.getProperty(ConsumerConfig.CLIENT_RACK_CONFIG)));
```

Comment:
This assertThat is missing the assertion invocation `.isEqualTo()`? 

Also I think the Properties object is discarded and our changes don't get trickled back up to us.

Reply:
Java arguments are pass-by-reference by default. So if you mutate an object in a method, it will be reflected after it's run.

Reply:
You're right, on both counts. Even though properties is passed by reference, a new one is instantiated and our values are copied into it.